src/libsnark.ml

```diff
@@ -1465,6 +1491,143 @@ module Bn128 = Make_full (struct
   let prefix = "camlsnark_bn128"
 end)
 
+module Make_proof_accessors (Prefix : sig
+  val prefix : string
+end)
+(Proof : Foreign_intf) (G1 : sig
+    include Deletable_intf
+
+    module Vector : Deletable_intf
+end)
+(G2 : Deletable_intf) =
+struct
+  open Prefix
+
+  let func_name = with_prefix prefix
+
+  let func name ret delete =
+    let stub = foreign (func_name name) (Proof.typ @-> returning ret) in
+    fun vk ->
+      let r = stub vk in
+      Caml.Gc.finalise delete r ; r
+
+  let a = func "a" G1.typ G1.delete
+
+  let b = func "b" G2.typ G2.delete
+
+  let c = func "c" G1.typ G1.delete
+end
+
+module Make_bowe_gabizon (M : sig
+  val prefix : string
+
+  module R1CS_constraint_system : sig
+    type t
+
+    val typ : t Ctypes.typ
+  end
+
+  module Field : sig
+    include Deletable_intf
+
+    val random : unit -> t
+
+    module Vector : sig
+      type t
+
+      val typ : t Ctypes.typ
+    end
+  end
+
+  module G1 : sig
+    include Deletable_intf
+
+    include Binable.S with type t := t
+
+    module Vector : Deletable_intf
+
+    val scale_field : t -> Field.t -> t
+  end
+
+  module G2 : sig
+    include Deletable_intf
+
+    include Binable.S with type t := t
+  end
+end) (H : sig
+  open M
+
+  val hash :
+       ?message:bool array
+    -> a:G1.t
+    -> b:G2.t
+    -> c:G1.t
+    -> delta_prime:G2.t
+    -> G1.t
+end) =
+struct
+  open M
+
+  include Make_proof_system_keys (struct
+    include M
+
+    let prefix = with_prefix M.prefix "bg"
+  end)
+
+  module Proof = struct
+    module Pre = struct
+      module Prefix = struct
+        let prefix = with_prefix M.prefix "bg_proof"
+      end
+
+      module T = struct
+        include Make_foreign (Prefix)
+      end
+
+      include T
+      include Make_proof_accessors (M) (T) (G1) (G2)
+
+      let delta_prime = func "delta_prime" G2.typ G2.delete
+
+      let verify_components ~a ~b ~c ~delta_prime ~z ~y_s key input =
+        let stub =
+          foreign
+            (func_name "verify_components")
```

Comment:
Should `Mnt4` and `Mnt6` have this function too? I can find it in `Bn128`, but the others have a `verify` function instead.

Reply:
good catch, thank you!